extensions/items/standalone.mjs

```diff
@@ -261,28 +281,72 @@ class Standalone {
    */
   async c2s_dropItem(playerID, mapCoord, facing, itemName) {
     /* player has not registered yet */
-    if (!this.items.has(playerID)) {
-      console.error('Player not initialized');
-      return;
+    if (!(playerID in this.items)) {
+      this.items[playerID] = {};
     }
 
-    /* update database */
-    let itemObj = this.items.get(playerID);
+    let itemObj = this.items[playerID];
     if (!itemName in itemObj) {
       console.error('No such item');
       return;
     }
+    if (!this.itemInfo[itemName].droppable) {
+      console.log("Item is not droppable");
+      return;
+    }
     itemObj[itemName].amount -= 1;
-    this.items.set(playerID, itemObj);
+    this.items[playerID, itemObj];
 
     /* update cell set */
-    /* TODO: calculate dropped coordinate. Currently, the dropped coordinate is the same as the player's map coordinate */
-    this.droppedItemCell.set(this.droppedItemIndex, { "x": mapCoord.x, "y": mapCoord.y, "w": 1, "h": 1 });
-    this.droppedItemInfo.set(this.droppedItemIndex, itemName);
-    this.droppedItemIndex++;
-    for (const mapName of this.helper.gameMap.getOriginalCellSetStartWith('').keys()) { // Get all available maps.
-      this.gameMap.updateDynamicCellSet(mapName, 'droppedItem' + itemName.toUpperCase(), Array.from(this.droppedItemCell.values()));
+    const mapSize = this.gameMap.getMapSize(mapCoord.mapName);
```

Comment:
This is assuming all coordinates are "droppable", not sure if we can assume that.